### DISCUSSION TITLE
Subscribe to Collection stopped working after changing some fields and filters, SvelteKit

### DISCUSSION THREAD

**Discussion by dimitarnikolovv** [pocketbase/pocketbase#4342] [Q&A]
Hello, recently I modified my notification system by adding some additional fields for filtering user groups. Before this change, I had a subscription to the notification collection that worked perfectly and displayed the newly created notifications in my SvelteKit app.
That's the code where I am subscribing to the collection and listening for a create action, if there is a create action I am invalidating the page so that the notification could be displayed in the app. The strange thing is that it works when running locally but not in production, even if I am using the same PocketBase backend.

```js
  onMount(async () => {
    pb.authStore.loadFromCookie(document.cookie);
    try {
      pb.authStore.isValid && (await pb.collection('users').authRefresh());
    } catch (_) {
      pb.authStore.clear();
    }

    if (data?.user) {
      await pb.collection('notifications').subscribe('*', async ({ record, action }) => {
        if (action === 'create') {
          await invalidateAll();
          notificationRedirect = record.redirects;
          newNotification = true;

          console.log(record);

          setTimeout(() => {
            newNotification = false;
            notificationRedirect = '';
          }, 4500);
        }
      });
    }
```

The problem is that after adding the fields that I am using for filtering the subscription stopped working and won't trigger the `InvalidateAll()` function. The notifications are shown after a full refresh and invoking the `pb.collection('notifications').getFullList()`

Here is the collection and its rules:
![Screenshot 2024-02-13 at 16 33 38](https://github.com/pocketbase/pocketbase/assets/49731322/9b6a7a6c-b52d-44a6-ab72-e779b602da5f)

![Screenshot 2024-02-13 at 16 33 51](https://github.com/pocketbase/pocketbase/assets/49731322/c95f890a-f662-4d8a-aee8-91cc78266422)

And this is the filter that I am using to retrieve the notifications:
```js
 await locals.pb.collection('notifications').getFullList({
        filter: `user="${locals.user.id}"||(mode="all"||mode="${
          locals?.subscribed ? 'subscribed' : 'non-subscribed'
        }" ${locals.user.admin ? '||mode="admins"' : ''})`,
        sort: '-created',
      });
```

**Comment by ganigeorgiev** [comment]
Other than making sure that you are authenticated before sending the `subscribe()` call, I don't know what else to suggest.

You can also start the executable with `--dev` and/or change the min log level from the _Admin UI > Logs_ settings to `-4` to see if there are any details about the fired realtime events.

If you are not able to resolve it, please try to create a minimal reprodible repo or at least export your collections configuration as json (_Admin UI > Settings > Export collections_) and I'll try to test it.

**Reply by dimitarnikolovv** [reply]
The `authRefresh()` and `if (data?.user)` ensure that there is an authenticated user. I already tried to debug the issue with the logs, but there is nothing wrong there too. 


Exported collections: 
[pb_schema (1).json](https://github.com/pocketbase/pocketbase/files/14268301/pb_schema.1.json)

**Reply by ganigeorgiev** [reply]
The SDK doesn't know what `data.user` is and the `authRefresh()` may fail and will clear the store.

I'll test with the provided schema a little bit later today.

**Reply by dimitarnikolovv** [reply]
Yeah, but if there is no user object the SDK call won't be made at all and that's on purpose, I already checked if the user is present while testing, and even if there is a user it won't send the `subscribe()` call.

**Reply by ganigeorgiev** [reply]
I've tried to test it locally and it works fine for me as far as I understand (I've tried with mode `all` and `admins` + updated the user flag). You can try to place the following static html in `pb_public/index.html`:

```html
<!DOCTYPE html />
<html>
<head>
    <meta charset="utf-8" />
    <title>Subscribe test</title>
</head>
<body>
    <script type="module">
        import PocketBase from "https://cdn.jsdelivr.net/gh/pocketbase/js-sdk@0.21.1/dist/pocketbase.es.mjs"

        const pb = new PocketBase("http://127.0.0.1:8090");

        await pb.collection("users").authWithPassword("test@example.com", "1234567890")

        await pb.collection("notifications").subscribe("*", (e) => {
            console.log("notifications:", e);
        })
    </script>
</body>
</html>
```
Navigate to `http://127.0.0.1:8090/` in your browser open the console and add a new notification for your user.

> but if there is no user object the SDK call won't be made at all and that's on purpose

Again, the SDK doesn't know what `data.user` is and when that will be updated. `pb.authStore.token` is the only important bit.

**Reply by ganigeorgiev** [reply]
If you are still not able to identify the culprit, provide details what are the user settings you are testing with and the notifications data you are creating.

**Reply by dimitarnikolovv** [reply]
> I've tried to test it locally and it works fine for me as far as I understand 

That's the main problem, locally it works perfectly, but the problem occurs only when it's deployed (Vercel). And I think that Pocketbase has nothing to do with this as it is deployed to a separate server, both the production and dev instances.
